### PR TITLE
Doc build bug?

### PR DESCRIPTION
**Pull request by mishig25**
# What does this PR do?

If you compare doc-build [transformers](https://github.com/huggingface/doc-build/tree/main/transformers/master/en) vs [datasets](https://github.com/huggingface/doc-build/tree/main/datasets/master/en), you see that transformers master version files are NOT being overwritten. (the should be overwritten).

| tsfms | dtsts |
|-------|-------|
| Screenshot 2022-03-09 at 16 16 49   | Screenshot 2022-03-09 at 16 18 19   |

When I compared build_documentation.yml between [tsfms](https://github.com/huggingface/transformers/blob/master/.github%2Fworkflows%2Fbuild_documentation.yml#L100) & [datasets](https://github.com/huggingface/datasets/blob/master/.github%2Fworkflows%2Fbuild_documentation.yml#L72-L79), I see
diff 
```
git stash && git pull && git stash apply &&
```

I don't see https://github.com/huggingface/doc-builder/pull/131 being applied when I go to https://huggingface.co/docs/transformers/master/en/model_doc/bert

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sgugger** [comment]
Removing this line will cause an error each time the doc-build-dev repo has been updated by another PR on Transformers or another library.

**Comment by sgugger** [comment]
Ugh, it does look like it prevents any update though, which is weird.

File: .github/workflows/build_documentation.yml [hunk 1]
```diff
@@ -97,7 +97,6 @@ jobs:
           cd doc-build &&
           if [[ `git status --porcelain` ]]; then 
             git add . &&
-            git stash && git pull && git stash apply &&
             git commit -m "Updated with commit ${{ github.sha }} \n\nSee: https://github.com/huggingface/transformers/commit/${{ github.sha }}" &&
             git push origin main
```
Comment (mishig25) [inline .github/workflows/build_documentation.yml:101]:
```suggestion
            git pull
            git push origin main
```

will this work?

Comment (sgugger) [reply .github/workflows/build_documentation.yml:101]: No, the pull will fail if we don't stash the changes. I will suggest another fix in a few minutes.

**Pull request closed without merging**